apps/agentic-chat/src/components/assistant-ui/BebopQuoteUI.tsx

```diff
@@ -31,65 +43,68 @@ export type BebopRateResult = {
 
 const BebopQuoteUI = makeAssistantToolUI<BebopRateArgs, BebopRateResult>({
   toolName: 'bebopRate',
-  render: ({ result }) => {
-    if (typeof result === 'string') return null;
-    if (!result) return null;
-    return (
-      <div
-        style={{
-          background: '#fff',
-          borderRadius: 20,
-          padding: 32,
-          maxWidth: 400,
-          margin: '0 auto',
-          boxShadow: '0 2px 16px rgba(0,0,0,0.08)',
-        }}
-      >
-        <h2 style={{ textAlign: 'center', fontWeight: 700, marginBottom: 32 }}>
-          Confirm Trade
-        </h2>
-        <div style={{ marginBottom: 24 }}>
-          <div style={{ color: '#888', fontWeight: 500, marginBottom: 8 }}>
-            Sell Amount
-          </div>
-          <div
-            style={{
-              background: '#f5f5f5',
-              borderRadius: 8,
-              padding: '16px 20px',
-              fontSize: 20,
-              display: 'flex',
-              alignItems: 'center',
-              justifyContent: 'space-between',
-              color: '#aaa',
-            }}
-          >
-            <span>{result.sellAmountCryptoPrecision}</span>
-            <span style={{ color: '#444', fontWeight: 600 }}>
-              {result.sellAsset.symbol}
-            </span>
-          </div>
-        </div>
-        <div>
-          <div style={{ color: '#888', fontWeight: 500, marginBottom: 8 }}>
-            Buy Amount
-          </div>
-          <div
-            style={{
-              background: '#f5f5f5',
-              borderRadius: 8,
-              padding: '16px 20px',
-              fontSize: 20,
-              color: '#aaa',
-            }}
-          >
-            <span>
-              {result.buyAmountCryptoPrecision} {result.buyAsset.symbol}
-            </span>
-          </div>
-        </div>
-      </div>
-    );
+  render: ({ status, result, args, isError }) => {
+    switch (status.type) {
+      case 'running':
+      case 'requires-action':
+      case 'incomplete':
```

Comment:
are there cases outside of `default` that we don't want to show this? only UI without the shimmer in the default case